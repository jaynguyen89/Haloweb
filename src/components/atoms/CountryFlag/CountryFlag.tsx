import React, { CSSProperties } from 'react';
import '/node_modules/flag-icons/css/flag-icons.min.css';
import useStyles from 'src/components/atoms/CountryFlag/styles';

// https://flagicons.lipis.dev/

interface ICountryFlag {
    isoCountryCode: string,
    variant?: 'square' | 'rectangle',
    center?: true,
    styles?: CSSProperties,
    className?: string,
}

const CountryFlag = ({
    isoCountryCode,
    variant = 'rectangle',
    center,
    styles: extraStyles,
    className: extraClasses,
}: ICountryFlag) => {
    const styles = useStyles();

    return (
        <span
            className={`fi fi-${isoCountryCode.toLowerCase()} ${variant === 'square' ? 'fis' : ''} ${center ? styles.center : undefined} ${extraClasses}`}
            style={extraStyles}
        />
    );
};

export default CountryFlag;
