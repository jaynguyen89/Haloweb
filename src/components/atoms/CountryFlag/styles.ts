import { makeStyles } from '@mui/styles';

const useStyles = makeStyles(() => ({
    center: {
        backgroundSize: 'contain',
        backgroundPosition: '50%',
        backgroundRepeat: 'no-repeat',
    },
}));

export default useStyles;
