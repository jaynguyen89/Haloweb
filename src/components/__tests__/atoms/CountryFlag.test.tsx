import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import CountryFlag from 'src/components/atoms/CountryFlag/CountryFlag';

configure({ adapter: new Adapter() });
describe('CountryFlag', () => {
    it('renders successfully', () => {
        const wrapper = shallow(<CountryFlag isoCountryCode='test' />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render CountryFlag with the given data', () => {
        const rectangle = shallow(<CountryFlag isoCountryCode='au' />);
        expect(rectangle.hasClass('fi-au')).toBeTruthy();
        expect(rectangle.hasClass('fis')).toBeFalsy();

        const square = shallow(<CountryFlag isoCountryCode='vn' variant='square' />);
        expect(square.hasClass('fi-vn')).toBeTruthy();
        expect(square.hasClass('fis')).toBeTruthy();
    });
});
