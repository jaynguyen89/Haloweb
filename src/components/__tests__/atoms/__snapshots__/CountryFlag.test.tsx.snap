// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryFlag renders successfully 1`] = `
<span
  className="fi fi-test  undefined undefined"
/>
`;
